### ISSUE TITLE
An addon to prevent the editor from "popping out" replaced blocks

### ISSUE THREAD

**Issue by penguinmoose** [ScratchAddons/ScratchAddons#5098]
### Why this would be helpful/solve a problem

I always need to replace reporter blocks in the editor. It is possible to do so by dragging the new reporter block(s) into the space of the reporter block(s) I want to replace, but that means that the old blocks will be "popped out" and makes the code very dirty.

### What this addon does

An addon that prevents this from happening would be great. Instead of the editor "popping out" replaced blocks, the blocks would simply be removed.

### Demonstration(s) and additional context

<img width="293" alt="Screen Shot 2022-09-10 at 5 04 58 PM" src="https://user-images.githubusercontent.com/67451414/189505920-a76722d3-b653-4afb-a77e-3087bf6fa43b.png">
<img width="266" alt="Screen Shot 2022-09-10 at 5 06 44 PM" src="https://user-images.githubusercontent.com/67451414/189505922-c60337d6-6d31-4bde-b947-33c5026f22d2.png">

**Comment by rgantzosonscratch** [comment]
This could get confusing and make mistakes even harder to fix. Maybe it would be better if you could hold down shift or something to make that happen?

**Comment by BroJac5246** [comment]
What happens when you misclick? That would be annoying. That said, something like this would be useful.

**Comment by Secret-chest** [comment]
> What happens when you misclick? That would be annoying. That said, something like this would be useful.

Undo: *exists*

**Comment by Secret-chest** [comment]
> Possible alternatives\nNo response

Use right-click > clean up + from the addon

**Comment by penguinmoose** [comment]
@Secret-chest So you mean that clean up blocks would remove these popped-out blocks?

**Comment by Secret-chest** [comment]
> @Secret-chest So you mean that clean up blocks would remove these popped-out blocks?

Clean-up blocks + removes any orphaned reporters.

**Comment by BroJac5246** [comment]
> > @Secret-chest So you mean that clean up blocks would remove these popped-out blocks?
> 
> Clean-up blocks + removes any orphaned reporters.

That could be pretty useful if we added the option to do that. Would it affect orphaned booleans too?

**Comment by CST1229** [comment]
> > > @Secret-chest So you mean that clean up blocks would remove these popped-out blocks?
> > 
> > 
> > Clean-up blocks + removes any orphaned reporters.
> 
> That could be pretty useful if we added the option to do that. Would it affect orphaned booleans too?

Clean up Blocks + (in the developer tools addon, by the way) already does that, and I think it also removes orphaned booleans because they're basically the same (I don't see why an addon that affects reporters shouldn't also affect booleans too).

**Comment by Secret-chest** [comment]
> > > > @Secret-chest So you mean that clean up blocks would remove these popped-out blocks?
> > > 
> > > 
> > > Clean-up blocks + removes any orphaned reporters.
> > 
> > That could be pretty useful if we added the option to do that. Would it affect orphaned booleans too?
> 
> Clean up Blocks + (in the developer tools addon, by the way) already does that, and I think it also removes orphaned booleans (I don't see why an addon that affects reporters shouldn't also affect booleans too).

booleans are special reporters

**Comment by penguinmoose** [comment]
> > @Secret-chest So you mean that clean up blocks would remove these popped-out blocks?
> 
> Clean-up blocks + removes any orphaned reporters.

What if I don’t want to clean up my blocks? Clean up blocks is pretty space-inefficient, so I rarely use it.

**Comment by penguinmoose** [comment]
@Secret-chest @CST1229 @BroJac5246 I think there can be a toggle switch somewhere in the editor to turn this option (to not pop out blocks) on and off.

**Comment by BroJac5246** [comment]
> @Secret-chest @CST1229 @BroJac5246 I think there can be a toggle switch somewhere in the editor to turn this option (to not pop out blocks) on and off.
> 
> To fix accidental replacement, when a block is replaced, there can be a temporary “undo” button somewhere in the editor, probably in the header.

I feel like the option to turn this on and off from the editor is a bit strange. Why wouldn't you want this to be on all of the time?

As for the undo button, that could probably be integrated with the one that Scratch provides, although I'm curious to see what everyone else thinks.

**Comment by penguinmoose** [comment]
> > @Secret-chest @CST1229 @BroJac5246 I think there can be a toggle switch somewhere in the editor to turn this option (to not pop out blocks) on and off.
> > To fix accidental replacement, when a block is replaced, there can be a temporary “undo” button somewhere in the editor, probably in the header.
> 
> I feel like the option to turn this on and off from the editor is a bit strange. Why wouldn't you want this to be on all of the time?
> 
> As for the undo button, that could _probably_ be integrated with the one that Scratch provides, although I'm curious to see what everyone else thinks.

True. Ok, if undo works, then why would accidental replacement be a problem?

**Comment by BroJac5246** [comment]
> True. Ok, if undo works, then why would accidental replacement be a problem?

It would probably just be an annoyance to consider.
